src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AutoInsert/AutoClosingTagOnAutoInsertProvider.cs

```diff
@@ -211,8 +212,10 @@ private static bool TryEnsureOwner_WorkaroundCompilerQuirks(int afterCloseAngleI
                 currentOwner = parentElement.StartTag.CloseAngle;
             }
         }
-        else if (currentOwner.Parent is MarkupTagHelperElementSyntax parentTagHelperElement &&
-            parentTagHelperElement.StartTag != null)
+        else if (currentOwner.Parent is MarkupTagHelperElementSyntax
+        {
+            StartTag: not null
+        } parentTagHelperElement)
```

Comment:
Before you say anything, yes the indentation is off here, but looks like its a Roslyn bug. Logged https://github.com/dotnet/roslyn/issues/67090